### ISSUE TITLE
 Base mainnet. Sync failed: couldn't set state to stateRoot

### ISSUE THREAD

**Issue by mikhail-khalizev** [NethermindEth/nethermind#6327]
**Description**
Base mainnet synchronization fails.

**Steps to Reproduce**
1. Run docker `nethermind/nethermind:1.23.0` with `--config=base-mainnet`
2. Run docker `us-docker.pkg.dev/oplabs-tools-artifacts/images/op-node:v1.3.1` with Base mainnet settings.

**Actual behavior**
Nethermind Logs:

```
05 Dec 14:39:40 [Merge.Plugin.Handlers.ForkchoiceUpdatedHandler] Received ForkChoice: Head: 26463 (0x38bbea...eb68a9), Safe: 26463 (0x38bbea...eb68a9), Finalized: 26281 (0x56fc0b...7f00fd) PayloadAttributes {Timestamp: 1686842275, PrevRandao: 0x7c3e06360b569051e67ef7513094380b69c9b5a03b7e430d4e19707c3570a2f8, SuggestedFeeRecipient: 0x4200000000000000000000000000000000000011, GasLimit: 30000000, NoTxPool: True, Transactions: 1} 
05 Dec 14:39:40 [JsonRpc.JsonRpcService] Error during method execution, request: ID 2383, engine_forkchoiceUpdatedV2({
  "headBlockHash": "0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9",
  "safeBlockHash": "0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9",
  "finalizedBlockHash": "0x56fc0bc3c93b600e7dac0c2975103681bde798e3c8b8846d65060ecfb47f00fd"
},{
  "timestamp": "0x648b2ba3",
  "prevRandao": "0x7c3e06360b569051e67ef7513094380b69c9b5a03b7e430d4e19707c3570a2f8",
  "suggestedFeeRecipient": "0x4200000000000000000000000000000000000011",
  "transactions": [
    "0x7ef90159a052c7bae87047e2755603938be2bfbd6eb24a619a2f3ec5997a869c7a0e27b76294deaddeaddeaddeaddeaddeaddeaddeaddead00019442000000000000000000000000000000000000158080830f424080b90104015d8eb900000000000000000000000000000000000000000000000000000000010ad12100000000000000000000000000000000000000000000000000000000648b2b5300000000000000000000000000000000000000000000000000000005d13c3300a0a93e56777a6b66b373b6ae583c42251966214dd7481138ba06fa814f1c53ce00000000000000000000000000000000000000000000000000000000000000040000000000000000000000005050f69a9786f081509234f1a7f4684b5e5b76c900000000000000000000000000000000000000000000000000000000000000bc00000000000000000000000000000000000000000000000000000000000a6fe0"
  ],
  "noTxPool": true,
  "gasLimit": "0x1c9c380"
}) Nethermind.Merge.Plugin.BlockProduction.EmptyBlockProductionException: Couldn't produce empty block: Setting state for processing block failed: couldn't set state to stateRoot 0xa9adc8eb3f0f920828237af738e7293d7d3c6d64985608176c30bf6a60ea920e
   at Nethermind.Optimism.OptimismPostMergeBlockProducer.PrepareEmptyBlock(BlockHeader parent, PayloadAttributes payloadAttributes) in /src/Nethermind/Nethermind.Optimism/OptimismPostMergeBlockProducer.cs:line 84
   at Nethermind.Merge.Plugin.BlockProduction.PayloadPreparationService.ProduceEmptyBlock(String payloadId, BlockHeader parentHeader, PayloadAttributes payloadAttributes) in /src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs:line 92
   at Nethermind.Merge.Plugin.BlockProduction.PayloadPreparationService.StartPreparingPayload(BlockHeader parentHeader, PayloadAttributes payloadAttributes) in /src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs:line 78
   at Nethermind.Merge.Plugin.Handlers.ForkchoiceUpdatedHandler.Handle(ForkchoiceStateV1 forkchoiceState, PayloadAttributes payloadAttributes, Int32 version) in /src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs:line 80
   at Nethermind.Merge.Plugin.EngineRpcModule.ForkchoiceUpdated(ForkchoiceStateV1 forkchoiceState, PayloadAttributes payloadAttributes, Int32 version) in /src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Paris.cs:line 61
   at Nethermind.Optimism.OptimismEngineRpcModule.engine_forkchoiceUpdatedV2(ForkchoiceStateV1 forkchoiceState, OptimismPayloadAttributes payloadAttributes) in /src/Nethermind/Nethermind.Optimism/OptimismEngineRpcModule.cs:line 34
   at Nethermind.JsonRpc.JsonRpcService.ExecuteAsync(JsonRpcRequest request, String methodName, ValueTuple`2 method, JsonRpcContext context) in /src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs:line 235
   at Nethermind.JsonRpc.JsonRpcService.ExecuteRequestAsync(JsonRpcRequest rpcRequest, JsonRpcContext context) in /src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs:line 115
   at Nethermind.JsonRpc.JsonRpcService.SendRequestAsync(JsonRpcRequest rpcRequest, JsonRpcContext context) in /src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs:line 105
```

**Expected behavior**
No error.

**Logs**
op-node logs:

```
lvl=info msg="Sync progress"                          reason="reconciled with L1"    l2_finalized=0x56fc0bc3c93b600e7dac0c2975103681bde798e3c8b8846d65060ecfb47f00fd:26281 l2_safe=0x86ec5ae4e13c54d697089457883df9b6e431891a0052b35aed2f515b19adaeb8:26462 l2_unsafe=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9:26463 l2_engineSyncTarget=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9:26463 l2_time=1,686,842,273 l1_derived=0x1f1b3fcccc8ea442919bd33c3d09565593b4d971884713358af18ea01d50d731:17486121
lvl=info msg="Found next batch"                       batch_timestamp=1,686,842,273 parent_hash=0x86ec5ae4e13c54d697089457883df9b6e431891a0052b35aed2f515b19adaeb8 batch_epoch=0xa0a93e56777a6b66b373b6ae583c42251966214dd7481138ba06fa814f1c53ce:17486113 txs=0
lvl=info msg="generated attributes in payload queue"  txs=1 timestamp=1,686,842,273
lvl=info msg="Sync progress"                          reason="reconciled with L1"    l2_finalized=0x56fc0bc3c93b600e7dac0c2975103681bde798e3c8b8846d65060ecfb47f00fd:26281 l2_safe=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9:26463 l2_unsafe=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9:26463 l2_engineSyncTarget=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9:26463 l2_time=1,686,842,273 l1_derived=0x1f1b3fcccc8ea442919bd33c3d09565593b4d971884713358af18ea01d50d731:17486121
lvl=info msg="Found next batch"                       batch_timestamp=1,686,842,275 parent_hash=0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9 batch_epoch=0xa0a93e56777a6b66b373b6ae583c42251966214dd7481138ba06fa814f1c53ce:17486113 txs=0
lvl=info msg="generated attributes in payload queue"  txs=1 timestamp=1,686,842,275
lvl=warn msg="Failed to share forkchoice-updated signal" state="&{HeadBlockHash:0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9 SafeBlockHash:0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9 FinalizedBlockHash:0x56fc0bc3c93b600e7dac0c2975103681bde798e3c8b8846d65060ecfb47f00fd}" attr="&{Timestamp:0x648b2ba3 PrevRandao:0x7c3e06360b569051e67ef7513094380b69c9b5a03b7e430d4e19707c3570a2f8 SuggestedFeeRecipient:0x4200000000000000000000000000000000000011 Withdrawals:<nil> Transactions:[0x7ef90159a052c7bae87047e2755603938be2bfbd6eb24a619a2f3ec5997a869c7a0e27b76294deaddeaddeaddeaddeaddeaddeaddeaddead00019442000000000000000000000000000000000000158080830f424080b90104015d8eb900000000000000000000000000000000000000000000000000000000010ad12100000000000000000000000000000000000000000000000000000000648b2b5300000000000000000000000000000000000000000000000000000005d13c3300a0a93e56777a6b66b373b6ae583c42251966214dd7481138ba06fa814f1c53ce00000000000000000000000000000000000000000000000000000000000000040000000000000000000000005050f69a9786f081509234f1a7f4684b5e5b76c900000000000000000000000000000000000000000000000000000000000000bc00000000000000000000000000000000000000000000000000000000000a6fe0] NoTxPool:true GasLimit:0x1c9c380}" err="Internal error"
lvl=warn msg="Derivation process temporary error"     attempts=1 err="engine stage failed: temp: temporarily cannot insert new safe block: failed to create new block via forkchoice: unrecognized rpc error: Internal error"
lvl=warn msg="Failed to share forkchoice-updated signal" state="&{HeadBlockHash:0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9 SafeBlockHash:0x38bbeabfb41bdbbe6c7acd5d659fda215bbf7408e943e3abfe2b940f69eb68a9 FinalizedBlockHash:0x56fc0bc3c93b600e7dac0c2975103681bde798e3c8b8846d65060ecfb47f00fd}" attr="&{Timestamp:0x648b2ba3 PrevRandao:0x7c3e06360b569051e67ef7513094380b69c9b5a03b7e430d4e19707c3570a2f8 SuggestedFeeRecipient:0x4200000000000000000000000000000000000011 Withdrawals:<nil> Transactions:[0x7ef90159a052c7bae87047e2755603938be2bfbd6eb24a619a2f3ec5997a869c7a0e27b76294deaddeaddeaddeaddeaddeaddeaddeaddead00019442000000000000000000000000000000000000158080830f424080b90104015d8eb900000000000000000000000000000000000000000000000000000000010ad12100000000000000000000000000000000000000000000000000000000648b2b5300000000000000000000000000000000000000000000000000000005d13c3300a0a93e56777a6b66b373b6ae583c42251966214dd7481138ba06fa814f1c53ce00000000000000000000000000000000000000000000000000000000000000040000000000000000000000005050f69a9786f081509234f1a7f4684b5e5b76c900000000000000000000000000000000000000000000000000000000000000bc00000000000000000000000000000000000000000000000000000000000a6fe0] NoTxPool:true GasLimit:0x1c9c380}" err="Internal error"
lvl=warn msg="Derivation process temporary error"     attempts=2 err="engine stage failed: temp: temporarily cannot insert new safe block: failed to create new block via forkchoice: unrecognized rpc error: Internal error"
```

**Comment by deffrian** [comment]
Hi @mikhail-khalizev, can you share full nethermind and op-node logs?

**Comment by mikhail-khalizev** [comment]
Since then I have restarted services and cleaned out the Nethermind database directory. And now I got another error in `op-node`:

```
lvl=warn msg="Derivation process temporary error"     attempts=3 err="engine stage failed: temp: temporarily cannot insert new safe block: nil id in forkchoice result when expecting a valid ID"
```
[2023-12-06-base.log.zip](https://www.dropbox.com/scl/fi/6r6o0xkeuagr02b7ihyn1/2023-12-06-base.log.zip?rlkey=wb64l8srruu4y5yrxto5zr19s)

---

I'll try to restart services again to reproduce the original error.

**Comment by mikhail-khalizev** [comment]
Reproduced the original error. Steps:
1. Start `nethermind` and `op-node` services.
2. Restart `nethermind` service after synchronization has started.

[2023-12-06 - base v2.log.zip](https://github.com/NethermindEth/nethermind/files/13573931/2023-12-06.-.base.v2.log.zip)

**Comment by deffrian** [comment]
@mikhail-khalizev Can you share all your config params?

**Comment by deffrian** [comment]
@mikhail-khalizev Please try to remove `--Sync.NetworkingEnabled false` as workaround. This flag is required to enable syncing. We fixed it in #6332

**Comment by mikhail-khalizev** [comment]
Removed `--Sync.NetworkingEnabled false`.
But the same result.

nethermind:
```
06 Dec 13:30:50 [Merge.Plugin.Handlers.ForkchoiceUpdatedHandler] Valid. ForkChoiceUpdated ignored - already in canonical chain. Request: ForkChoice: Head: 238963 (0xc243dc...c00c2b), Safe: 238963 (0xc243dc...c00c2b), Finalized: 238963 (0xc243dc...c00c2b) PayloadAttributes {Timestamp: 1687267275, PrevRandao: 0xdc38b8b801ec4603cda967525601ff06633f45c4d3726fd922b6804b124fe895, SuggestedFeeRecipient: 0x4200000000000000000000000000000000000011, GasLimit: 30000000, NoTxPool: True, Transactions: 1}. 
06 Dec 13:30:53 [HealthChecks.HealthChecksPlugin] No incoming messages from the consensus client that is required for sync. 
06 Dec 13:30:58 [HealthChecks.HealthChecksPlugin] No incoming messages from the consensus client that is required for sync. 
06 Dec 13:31:00 [Merge.Plugin.Handlers.ForkchoiceUpdatedHandler] Received ForkChoice: Head: 238963 (0xc243dc...c00c2b), Safe: 238963 (0xc243dc...c00c2b), Finalized: 238963 (0xc243dc...c00c2b) PayloadAttributes {Timestamp: 1687267275, PrevRandao: 0xdc38b8b801ec4603cda967525601ff06633f45c4d3726fd922b6804b124fe895, SuggestedFeeRecipient: 0x4200000000000000000000000000000000000011, GasLimit: 30000000, NoTxPool: True, Transactions: 1} 
06 Dec 13:31:00 [Merge.Plugin.Handlers.ForkchoiceUpdatedHandler] Valid. ForkChoiceUpdated ignored - already in canonical chain. Request: ForkChoice: Head: 238963 (0xc243dc...c00c2b), Safe: 238963 (0xc243dc...c00c2b), Finalized: 238963 (0xc243dc...c00c2b) PayloadAttributes {Timestamp: 1687267275, PrevRandao: 0xdc38b8b801ec4603cda967525601ff06633f45c4d3726fd922b6804b124fe895, SuggestedFeeRecipient: 0x4200000000000000000000000000000000000011, GasLimit: 30000000, NoTxPool: True, Transactions: 1}. 
06 Dec 13:31:03 [HealthChecks.HealthChecksPlugin] No incoming messages from the consensus client that is required for sync. 
06 Dec 13:31:08 [HealthChecks.HealthChecksPlugin] No incoming messages from the consensus client that is required for sync. 
06 Dec 13:31:10 [Merge.Plugin.Handlers.ForkchoiceUpdatedHandler] Received ForkChoice: Head: 238963 (0xc243dc...c00c2b), Safe: 238963 (0xc243dc...c00c2b), Finalized: 238963 (0xc243dc...c00c2b) PayloadAttributes {Timestamp: 1687267275, PrevRandao: 0xdc38b8b801ec4603cda967525601ff06633f45c4d3726fd922b6804b124fe895, SuggestedFeeRecipient: 0x4200000000000000000000000000000000000011, GasLimit: 30000000, NoTxPool: True, Transactions: 1}
```

op-node:
```
t=2023-12-06T13:29:49+0000 lvl=warn msg="did not finish previous block building, starting new building now" prev_onto=0xc243dc5d4dc4f0612f62e5c76e8bf0d58ec0239c222886f7aaba953cb4c00c2b:238963 prev_payload_id=0x5153097668668896 new_onto=0xc243dc5d4dc4f0612f62e5c76e8bf0d58ec0239c222886f7aaba953cb4c00c2b:238963
t=2023-12-06T13:29:49+0000 lvl=warn msg="Derivation process temporary error"     attempts=31 err="engine stage failed: temp: temporarily cannot insert new safe block: nil id in forkchoice result when expecting a valid ID"
t=2023-12-06T13:29:59+0000 lvl=warn msg="did not finish previous block building, starting new building now" prev_onto=0xc243dc5d4dc4f0612f62e5c76e8bf0d58ec0239c222886f7aaba953cb4c00c2b:238963 prev_payload_id=0x5153097668668896 new_onto=0xc243dc5d4dc4f0612f62e5c76e8bf0d58ec0239c222886f7aaba953cb4c00c2b:238963
t=2023-12-06T13:29:59+0000 lvl=warn msg="Derivation process temporary error"     attempts=32 err="engine stage failed: temp: temporarily cannot insert new safe block: nil id in forkchoice result when expecting a valid ID"
```
